crates/shared/src/price_estimation/trade_verifier.rs

```diff
@@ -131,43 +155,46 @@ impl TradeVerifier {
             ..Default::default()
         };
 
-        // Set up helper contracts impersonating trader and solver.
-        let mut overrides = hashmap! {
-            verification.from => StateOverride {
-                code: Some(deployed_bytecode!(Trader)),
-                ..Default::default()
-            },
-            solver.address() => StateOverride {
-                code: Some(deployed_bytecode!(Solver)),
-                ..Default::default()
-            },
-        };
-
-        let trader_impl = self
-            .code_fetcher
-            .code(verification.from)
+        let overrides = self
+            .prepare_state_overrides(&solver, verification, trade)
             .await
-            .context("failed to fetch trader code")
             .map_err(Error::SimulationFailed)?;
-        if !trader_impl.0.is_empty() {
-            // Store `owner` implementation so `Trader` helper contract can proxy to it.
-            overrides.insert(
-                Self::TRADER_IMPL,
-                StateOverride {
-                    code: Some(trader_impl),
-                    ..Default::default()
-                },
-            );
-        }
 
         let block = self.block_stream.borrow().number;
         let output = self
             .simulator
             .simulate(call, overrides, Some(block))
             .await
             .context("failed to simulate quote")
-            .map_err(Error::SimulationFailed)?;
-        let summary = SettleOutput::decode(&output, query.kind)
+            .map_err(Error::SimulationFailed);
+
+        // TODO remove when quoters stop signign RFQ orders for `tx.origin: 0x0000`
```

Comment:
Can you create an issue for this and reference this here so we can easily find all the places we need to change when we get to this point?